src/meta-srv/src/procedure/region_failover/activate_region.rs

```diff
@@ -37,13 +37,18 @@ use crate::service::mailbox::{Channel, MailboxReceiver};
 #[derive(Serialize, Deserialize, Debug)]
 pub(super) struct ActivateRegion {
     candidate: Peer,
+    // If the meta leader node dies during the execution of the procedure,
+    // the new leader node needs to remark the failed region as "inactive"
+    // to prevent it from renewing the lease.
+    remark_inactive_region: bool,
```

Comment:
I think it might be easier if we `register_inactive_region` when the failover procedure is recovered from `ActivateRegion` state. (The recovery of failover procedure is done in `RegionFailoverProcedure::from_json`.)